python/packages/autogen-ext/src/autogen_ext/agents/mcp/_agent.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import Any, Awaitable, Callable, Dict, List, Sequence
+
+from autogen_agentchat.agents import AssistantAgent
+from autogen_agentchat.base import Handoff as HandoffBase
+from autogen_core.memory import Memory
+from autogen_core.model_context import ChatCompletionContext
+from autogen_core.models import ChatCompletionClient
+from autogen_core.tools import BaseTool, Workbench
+from pydantic import BaseModel
+
+from mcp import types as mcp_types
+
+from ...tools.mcp._base import ElicitorTypes
+from ...tools.mcp._host import McpWorkbenchHost
+
+
+class McpAssistantAgent(AssistantAgent, McpWorkbenchHost):
+    """An assistant agent that combines AssistantAgent functionality with MCP workbench hosting.
+
+    This agent inherits from both AssistantAgent and McpWorkbenchHost, providing:
+    - Full AssistantAgent capabilities (chat, tools, handoffs, memory, etc.)
+    - MCP workbench hosting (sampling, elicitation, listing roots)
+
+    The agent can serve as an MCP workbench host while maintaining all the standard
+    assistant agent functionality.
```

Comment:
A few doc-string examples to show how this can be used would be good. Be sure to include an MCP server that actually performs sampling, elicitation or listing roots actions.